sonarts-core/src/utils/areEquivalent.ts

```diff
@@ -32,7 +33,17 @@ const COMPARED_BY_TEXT = new Set([
   ts.SyntaxKind.JsxText,
 ]);
 
-export default function areEquivalent(first: ts.Node | ts.Node[], second: ts.Node | ts.Node[]): boolean {
+/**
+ *
+ * @param first
+ * @param second
+ * @param ignoreLiterals when true, two nodes are considered equivalent even if string or numeric literals have different value
+ */
+export default function areEquivalent(
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this function to reduce its Cognitive Complexity from 17 to the 15 allowed. ([typescript:S3776](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3776))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&pullRequest=625&issues=AWM_0mTiS2S-MqecJ7NA&open=AWM_0mTiS2S-MqecJ7NA)

[ISSUE_KEY]: AWM_0mTiS2S-MqecJ7NA